scm/src/GFS_typedefs.F90

```diff
@@ -3235,6 +3459,30 @@ subroutine control_initialize (Model, nlunit, fn_nml, me, master, &
     Model%do_tofd          = do_tofd
 
     Model%lssav            = lssav
+    !
+    if (naux2d>naux2dmax) then
+      write(0,*) "Error, number of requested auxiliary 2d arrays exceeds the maximum defined in GFS_typedefs.F90"
+      stop
+    endif
+    if (naux3d>naux3dmax) then
+      write(0,*) "Error, number of requested auxiliary 3d arrays exceeds the maximum defined in GFS_typedefs.F90"
+      stop
+    endif
+    Model%naux2d           = naux2d
+    Model%naux3d           = naux3d
+    if (Model%naux2d>0) then
+        allocate(Model%aux2d_time_avg(1:naux2d))
+        Model%aux2d_time_avg(1:naux2d) = aux2d_time_avg(1:naux2d)
+    end if
+    if (Model%naux3d>0) then
+        allocate(Model%aux3d_time_avg(1:naux3d))
+        Model%aux3d_time_avg(1:naux3d) = aux3d_time_avg(1:naux3d)
+    end if
+    if (any(aux2d_time_avg) .or. any(aux3d_time_avg)) then
+      write(0,*) "Error, the SCM has not implemented time averaging of diagnostics in GFS_typedefs.F90"
+      stop
+    end if
+    
```

Comment:
How is the time averaging done in SCM? The tendencies are accumulated in ccpp-physics. Are you subtracting the previous accumulated value after each time step?

Reply:
diag_phys_zero is called every time step and right before it is written out, these accumulated change variables are divided by the timestep to get the tendencies.